### PR TITLE
[IA-2675] make application concurrency configurable

### PR DESCRIPTION
**Pull request by rtitle**
https://broadworkbench.atlassian.net/browse/IA-2675

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]
👍

**Comment by rtitle** [comment]
Perf test running now..

**Comment by rtitle** [comment]
Don't think this really helped.. reverted back to previous value. Still think it's nice to make configurable.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala [hunk 1]
```diff
@@ -449,7 +449,8 @@ object Boot extends IOApp {
       // Set up k8s and helm clients
       kubeService <- org.broadinstitute.dsde.workbench.google2.KubernetesService
         .resource(Paths.get(pathToCredentialJson), gkeService, blocker, semaphore)
-      helmClient = new HelmInterpreter[F](blocker, semaphore)
+      helmConcurrency <- Resource.eval(Semaphore[F](50L))
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala]: Wondering if we'd get better throughput if helm had its own permits

**Comment by rtitle** [comment]
Actually seeing some different results at smaller concurrency levels for helm. I'm starting to think that the Galaxy `helm install` is just really chatty and we're limited by network speed on the back Leo node.

- concurrency of 255+: all 100 apps get installed at the same time, wait time ~45 mins for all helm installs to complete
- concurrency of 50: app installs are a bit more staggered, avg wait time ~10-15 mins per helm install
- concurrency of 10: app installs even more staggered, wait time ~2 mins per helm install

So I think a lower concurrency for helm probably increases throughput overall, since it staggers the `helm install`s and allows Galaxy to start up on some apps while helm is running on others (letting more things happen in parallel).

**Comment by rtitle** [comment]
100% test pass with concurrency of 10: http://34.70.158.48:8666/bvdp-daily/. The dev run last night was about 81% passing. Gonna try concurrency of 20 next.

**Pull request merged**